editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/AdvancedBeanPropertiesSection.java

```diff
@@ -0,0 +1,300 @@
+import org.fusesource.ide.camel.model.service.core.util.CamelComponentUtils;
+import org.fusesource.ide.foundation.core.util.Strings;
+
+/**
+ * @author bfitzpat
+ */
+public class AdvancedBeanPropertiesSection extends FusePropertySection {
+
+	/**
+	 * 
+	 * @param folder
+	 */
+	@Override
+	protected void createContentTabs(CTabFolder folder) {
+		List<Parameter> props = new ArrayList<>();
+		List<String> tabsToCreate = new ArrayList<>();
+		tabsToCreate.add(GROUP_COMMON);
+		Parameter idParam = BeanConfigUtil.createParameter(CamelBean.PROP_ID, String.class.getName());
+		idParam.setRequired("true");
+		props.add(idParam);
+		Parameter classParam = BeanConfigUtil.createParameter(CamelBean.PROP_CLASS, String.class.getName());
+		classParam.setRequired("true");
+		props.add(classParam);
+		props.add(BeanConfigUtil.createParameter(CamelBean.PROP_SCOPE, String.class.getName()));
+		props.add(BeanConfigUtil.createParameter(CamelBean.PROP_DEPENDS_ON, String.class.getName()));
+		props.add(BeanConfigUtil.createParameter(CamelBean.PROP_INIT_METHOD, String.class.getName()));
+		props.add(BeanConfigUtil.createParameter(CamelBean.PROP_DESTROY_METHOD, String.class.getName()));
+
+		props.sort(new ParameterPriorityComparator());
+
+		for (String group : tabsToCreate) {
+			CTabItem contentTab = new CTabItem(this.tabFolder, SWT.NONE);
+			contentTab.setText(Strings.humanize(group));
+
+			Composite page = this.toolkit.createComposite(folder);
+			page.setLayout(new GridLayout(4, false));
+
+			generateTabContents(props, page);
+
+			contentTab.setControl(page);
+
+			this.tabs.add(contentTab);
+		}
+	}
+
+	private IJavaProject getJavaProject() {
+		IProject project = CamelUtils.project();
+		if (project != null) {
+			IJavaProject jproject = JavaCore.create(project);
+			if (jproject.exists()) {
+				return jproject;
+			}
+		}
+		return null;
+	}
+
+	private IStatus validateMethod(Object value) {
+		Object control = modelMap.get(CamelBean.PROP_CLASS);
+		if (control != null) {
+			String className = (String) control;
+			String methodName = (String) value;
+			if (!Strings.isEmpty(methodName)) {
+				IJavaProject jproject = getJavaProject();
+				if (jproject != null) {
+					try {
+						IType foundClass = jproject.findType(className);
+						boolean foundMethod = BeanConfigUtil.hasMethod(methodName, foundClass);
+						if (!foundMethod) {
+							return ValidationStatus
+									.error("Method " + methodName + " must exist in class " + className + ".");
+						} else {
+							return ValidationStatus.ok();
+						}
+					} catch (JavaModelException e) {
+						CamelEditorUIActivator.pluginLog().logError("Error validating method for class " + className,
+								e);
+					}
+				}
+			}
+		}
+		return ValidationStatus.ok();
+	}
+
+	/**
+	 * 
+	 * @param props
+	 * @param page
+	 * @param ignorePathProperties
+	 * @param group
+	 */
+	protected void generateTabContents(List<Parameter> props, final Composite page) {
```

Comment:
it is the last one reported as critical.
I will take a look to get rid of the last reported as critical.